source/extensions/filters/http/decompressor/decompressor_filter.cc

```diff
@@ -72,7 +73,15 @@ Http::FilterHeadersStatus DecompressorFilter::decodeHeaders(Http::RequestHeaderM
   //      the upstream that this hop is able to decompress responses via the Accept-Encoding header.
   if (config_->responseDirectionConfig().decompressionEnabled() &&
       config_->requestDirectionConfig().advertiseAcceptEncoding()) {
-    headers.appendInline(accept_encoding_handle.handle(), config_->contentEncoding(), ",");
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.append_to_accept_content_encoding_only_once")) {
+      std::string newAcceptEncodingHeader = Http::HeaderUtility::addEncodingToAcceptEncoding(
```

Comment:
nit: naming convention has local vars as "new_accept_encoding_header".

Also this can be const.

Reply:
Fixed.